### PR TITLE
Add 'plotly_noembed' cargo feature

### LINKED ISSUE #176: plotly.min.js bloats binary size
Templates like [this one](https://github.com/igiagkiozis/plotly/blob/master/plotly/templates/plot.html#L14) include `plotly.min.js` such that static plots can be output correctly. For binaries using plotly that only output plots using the CDN distributed version of plotly, this unnecessarily increases the binary size.

Would it make sense to add a Cargo feature that allows to skip the inclusion of the minified plotly version (~2.5 MB)? For example `plotly-vendoredjs` with a default of `true` that can be set to `false`?

Or is there already a way to accomplish the desired behavior of not unnecessarily including the minified plotly version into the __TEXT,__const section of the binary?

### PR DESCRIPTION
**Pull request by sharkdp**
This adds a new `plotly_noembed` feature that can be used to reduce the sizes of binaries by not embedding `plotly.min.js`.

Since askama can not handle compile-time flags, we had to duplicate the `plot.html` and `static_plot.html` templates, where the new `*_noembed.html` version only contains the CDN-variant.

The `use_local_plotly` option has been feature-gated such that it is not available when using the new feature.

I verified that this works by building an application that depends on plotly.rs (including optimization and LTO in all cases):

    no plotly:   4.96 MiB (5,201,920 bytes)
    master:      8.96 MiB (9,400,320 bytes)
    this branch: 5.44 MiB (5,701,632 bytes)

The difference between this branch and master is 3,698,688 bytes, which is very close to the size of `plotly.min.js` (3,686,400 bytes). Just as expected.

closes #176

### CODE REVIEW

File: plotly/src/plot.rs [hunk 1]
```diff
@@ -29,6 +38,17 @@ struct StaticPlotTemplate<'a> {
...
+#[cfg(feature = "plotly_noembed")]
+#[derive(Template)]
+#[template(path = "static_plot_noembed.html", escape = "none")]
+#[cfg(not(target_family = "wasm"))]
+struct StaticPlotTemplate<'a> {
+    plot: &'a Plot,
+    format: ImageFormat,
+    width: usize,
+    height: usize,
+}
```
Comment (sharkdp) [inline plotly/src/plot.rs:50]: Not sure if there is a way to do this without having to duplicate the whole struct.

**Comment by andrei-ng** [comment]
@sharkdp , thank you for submitting the PR. 
I need a bit more time to review it.

**Comment by andrei-ng** [comment]
Hi @sharkdp,

I had a chance to look into the PR today. 

Indeed, duplicating the template files is annoying. I have looked into what alternatives we have. TBH, I thought about  it and I think the embedding of the `plotly.min.js` should be an opt-in feature and not an opt-out since it blows up the library size. Although I might risk upsetting some users, I am going to propose that we make it an `opt-in` . 

Also using the if/else in the template as it was originally implemented causes this issue of not being able to use the Rust compile time configuration as you pointed out in the #176 . We might be able to bypass that now since thanks to @GuillaumeGomez , we switched to `rinja` in #277, which comes with some really cool features and integrations of templates with Rust code. 

I propose for the moment a simpler approach, where we always use CDN and we bake in directly the local version only with the opt-in flag. I am going to push a commit on top of yours with these changes. Would be interested to hear your opinion.

**Review by andrei-ng — APPROVED** [review]

**Comment by sharkdp** [comment]
Awesome. I just tested it and it still works as expected. I'll integrate it into Numbat as soon as the release is out. Thank you :+1:

**Comment by andrei-ng** [comment]
I'll make a new release tomorrow.

**Comment by andrei-ng** [comment]
> Awesome. I just tested it and it still works as expected. I'll integrate it into Numbat as soon as the release is out. Thank you 👍

FYI, released 0.10.0 version just now.

**Comment by sharkdp** [comment]
Thank you! It's not integrated into Numbat :+1:

**Pull request merged**